File: problems/Codeforces/520A/solution.rs
use std::io::{self, Read};

fn main() -> io::Result<()> {
    let mut input = Vec::with_capacity(1 << 7);
    io::stdin().lock().read_to_end(&mut input)?;
    let mut count = 0u32;
    for &c in input
        .split(|e| e.is_ascii_whitespace())
        .filter(|s| !s.is_empty())
        .nth(1)
        .unwrap()
    {
        count |= 1 << ((c | 0x20) - b'a');
    }
    print!("{}", if count == (1 << 26) - 1 { "YES" } else { "NO" });
    Ok(())
}

File: problems/codeforces/520A/solution.cpp
#include <iostream>

int main() {
    using namespace std;
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    string n,s;
    cin >> n >> s;
    uint32_t count = 0;
    for(char& c : s) {
        count |= (1 << ((c | 0x20) - 'a'));
    }
    cout << (count == (1 << 26) - 1 ? "YES" : "NO");
    return 0;
}
